src/contracts/pods/EigenPodManager.sol

```diff
@@ -138,117 +121,98 @@ contract EigenPodManager is
-        );
-
-        uint32 limboStartBlock = _podOwnerUndelegationLimboStatus[msg.sender].startBlock;
-        require(
-            slasher.canWithdraw(
-                _podOwnerUndelegationLimboStatus[msg.sender].delegatedAddress,
-                limboStartBlock,
-                middlewareTimesIndex
-            ),
-            "EigenPodManager.exitUndelegationLimbo: shares in limbo are still slashable"
-        );
-
-        // enforce minimum delay lag
-        require(
-            limboStartBlock + strategyManager.withdrawalDelayBlocks() <= block.number,
-            "EigenPodManager.exitUndelegationLimbo: withdrawalDelayBlocks period has not yet passed"
-        );
-
-        // delete the pod owner's undelegation limbo details
-        delete _podOwnerUndelegationLimboStatus[msg.sender];
-
-        // emit event
-        emit UndelegationLimboExited(msg.sender);
-
-        // either withdraw the funds entirely from EigenLayer
-        if (withdrawFundsFromEigenLayer) {
-            // store the pod owner's current share amount in memory, for gas efficiency
-            uint256 currentPodOwnerShares = podOwnerShares[msg.sender];
-            // remove the shares from the podOwner and reduce the `withdrawableRestakedExecutionLayerGwei` in the pod
-            podOwnerShares[msg.sender] = 0;
-            _decrementWithdrawableRestakedExecutionLayerGwei(msg.sender, currentPodOwnerShares);
-            // withdraw through the ETH from the EigenPod
-            _withdrawRestakedBeaconChainETH(msg.sender, msg.sender, currentPodOwnerShares);
-            // or else return the "shares" to the delegation system
-        } else {
-            delegationManager.increaseDelegatedShares(msg.sender, beaconChainETHStrategy, podOwnerShares[msg.sender]);
+    function addShares(
+        address podOwner,
+        uint256 shares
+    ) external onlyDelegationManager returns (uint256) {
+        require(podOwner != address(0), "EigenPodManager.addShares: podOwner cannot be zero address");
+        require(int256(shares) >= 0, "EigenPodManager.addShares: shares cannot be negative");
+
+        int256 currentPodOwnerShares = podOwnerShares[podOwner];
+        int256 updatedPodOwnerShares = currentPodOwnerShares + int256(shares);
+        podOwnerShares[podOwner] = updatedPodOwnerShares;
+
+        return uint256(_calculateChangeInDelegatableShares({sharesBefore: currentPodOwnerShares, sharesAfter: updatedPodOwnerShares}));
+    }
+
+    /// @notice Used by the DelegationManager to complete a withdrawal, sending tokens to some destination address
+    /// @dev Prioritizes decreasing the podOwner's share deficit, if they have one
+    function withdrawSharesAsTokens(
+        address podOwner, 
+        address destination, 
+        uint256 shares
+    ) external onlyDelegationManager {
+        require(int256(shares) >= 0, "EigenPodManager.withdrawSharesAsTokens: shares cannot be negative");
```

Comment:
For consistency with `addShares` and `removeShares`, let's make the same "zero checks" for `podOwner` and `destination` here

Reply:
good suggestion, added in https://github.com/Layr-Labs/eigenlayer-contracts/pull/222/commits/e9782d99ba964787727d4fdf7d83980630f493bc